File: only code/1419/1419.py
import sys
def finder(l,r,k):
    if k == 2:
        result = r - max(3,l) + 1
    elif k == 3:
        start = max(6,l)
        end = r
        result = end // 3 - (start + 2) // 3 + 1
    elif k == 4:
        start = max(14,l)
        end = r
        result = end // 2 - (start+1) // 2 + 1
        if l <= 10:
            result += 1
    elif k == 5:
        start = max(15,l)
        end = r
        result = end // 5 - (start + 4) // 5 + 1
    return max(0,result)
l, r, k = map(int, [sys.stdin.readline() for _ in range(3)])
print(finder(l,r,k))

File: only code/1419/1419.cpp
#include <iostream>
using namespace std;

int finder(long long l, long long r, int k) {
    int t = k * (k - 1) / 2;
    int max_d = (r - k) / t + 1;
    long result = 0;

    for (int i = 1; i < max_d; ++i) {
        int max_x = (r - i * t) / k;
        int min_x = (l - i * t + k - 1) / k;

        for (int j = max(min_x, 1); j < max_x; ++j) {
            long long s = i * t + j * k;
            if (s > r) break;
            if (s >= l && s <= r) {
                result++;
            }
        }
    }

    return result;
}

int main() {
    long long l, r;
    int k;
    cin >> l >> r >> k;
    cout << finder(l, r, k) << endl;
    return 0;
}
